### PR TITLE
Basic VSCode configuration for building and debugging

### PR DESCRIPTION
**Pull request by lafriks**
I know that there are many options for IDE/code editors developing Go but it would make easier for new developers to get going by providing at least basic configuration for building and debugging out of box.

### CODE REVIEW

**Comment by lunny** [comment]
LGTM, I'm using VSCode.

**Comment by Morlinest** [comment]
I am using vscode too, but don't think any IDE setup should be included in project codebase. You can add it to README or create wiki page about it.

**Comment by bkcsoft** [comment]
Yeah, add this to `scripts/` instead (and maybe change that to `support/` ?)

File: .vscode/tasks.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      "command": "go",
+      "group": "build",
+      "presentation": {
+        "echo": true,
+        "reveal": "always",
+        "focus": false,
+        "panel": "shared"
+      },
+      "args": ["build", "-o", "gitea", "${workspaceRoot}/main.go"],
+      "windows": {
```
Comment (cez81) [inline .vscode/tasks.json]: Doesn't work for me on Windows. Looks like it's combing both sets of args.
```
> Executing task: go build -o gitea c:\Users\Jonas\go\src\code.gitea.io\gitea/main.go build -o gitea.exe c:\Users\Jonas\go\src\code.gitea.io\gitea/main.go <

named files must be .go files
The terminal process terminated with exit code: 1
```

**Comment by pgaskin** [comment]
LGTM, I'm using VSCode on Debian, and this works for me.

**Comment by lafriks** [comment]
@cez81 fixed for windows

**Comment by lafriks** [comment]
@Morlinest @bkcsoft my idea was for new developers to get going faster and it does not mess anything if you are other IDE/editor but off course I can move it to `contrib` also but that would not be much of the help as most probably anyone rarely looks in that folder :)

**Comment by cez81** [comment]
@lafriks Works 👍 

I think it can be moved to contrib folder but should be accompanied with support docs to get people started.

**Comment by lafriks** [comment]
@sapk @Morlinest @bkcsoft done

**Comment by Morlinest** [comment]
@lafriks Thanks for all your work (I will probably use it too :D).

I want to share my opinion:
I am glad you want to help new developers and I appreciate that. Great help would be updated readme with manual/examples as this is first document anyone sees and reads (and there is nothing about how to run, develop or debug gitea - "contributors guide" is mentioned, but it doesn't contain any usefull command for me). You can add example configuration for any ide. The best way for me is to add it to different folder (as your updated PR) or documentation like wiki or directly to readme file. Yes, it will not conflict with other IDE if you add it to project, but what about people that want to use vscode with their own setup?

Another thing, there is makefile for help with everything and I am missing any documentation for that. This (documentation for makefile and what commands should I use and when) would be for me best first help.

**Comment by lafriks** [comment]
Make LG-TM work

**Pull request merged**